src/controller/CHIPDeviceController.cpp

```diff
@@ -2740,6 +2747,18 @@ DeviceCommissioner::SendCommissioningCommand(DeviceProxy * device, const Request
                                 onFailureCb, NullOptional, timeout, (!fireAndForget) ? &mInvokeCancelFn : nullptr);
 }
 
+template <typename AttrType>
+CHIP_ERROR DeviceCommissioner::SendCommissioningWriteRequest(DeviceProxy * device, EndpointId endpoint, ClusterId cluster,
+                                                             AttributeId attribute, const AttrType & requestData,
+                                                             WriteResponseSuccessCallback successCb,
+                                                             WriteResponseFailureCallback failureCb)
+{
+    auto onSuccessCb = [this, successCb](const app::ConcreteAttributePath & aPath) { successCb(this); };
+    auto onFailureCb = [this, failureCb](const app::ConcreteAttributePath * aPath, CHIP_ERROR aError) { failureCb(this, aError); };
+    return WriteAttribute(device->GetSecureSession().Value(), endpoint, cluster, attribute, requestData, onSuccessCb, onFailureCb,
```

Comment:
This write needs to be cancelable, so we don't get problems if someone stops the commissioning and starts a new one while this is in flight, no?

Reply:
Write operation here is used for writing `interfaceEnabled` attribute of commissionee to false. The commissionee will revert network to the previous state and reset  `interfaceEnabled` attribute to true, when fail-safe timer is expired or commissionee has been rebooted with the fail-safe armed. 
BUT I am not clear that how to make the write operation cancelable. Any specific suggestions?

Reply:
See how command invokes work in CHIPDeviceController.

Basically: if commissioning is stopped, we need to ensure that callbacks from this write do not reach the CHIPDeviceController after that point.

Reply:
Added write request cancel function.